tmt/uuid.py

```diff
@@ -0,0 +1,119 @@
+import re
+from uuid import uuid4
+
+import fmf
+
+import tmt.base
+
+from .utils import log
+
+ID_FMF_ITEM = "id"
+
+
+class IDError(Exception):
+    pass
+
+
+class IDLeafError(IDError):
+    pass
+
+
+def dereference_node(node):
+    """
+    find fmf items inside node
+    """
+    if isinstance(node, fmf.Tree):
+        return node
+    elif isinstance(node, tmt.base.Node):
+        return node.node
+    else:
+        raise ValueError(f"{node} is not part of fmf data")
+
+
+def id_location(node, name=ID_FMF_ITEM):
+    """
+    Return location of ID with path
+    """
+    node = dereference_node(node)
+    identifier = node.data.get(name)
+    if identifier is None:
+        raise IDError("ID not defined")
+    current_node = node
+    while True:
+        parent = current_node.parent
+        if parent:
+            if current_node.get(name) == parent.get(name):
+                current_node = parent
+                continue
+            else:
+                return current_node
+        else:
+            if current_node.get(name):
+                return current_node
+            else:
+                break
+    return None
+
+
+def is_id_defined_in_leaf(node):
+    """
+    Is ID defined in leaf node
+    """
+    try:
+        location = id_location(node).name
+        if node and dereference_node(node).name == location:
+            return True
+        return False
+    except IDError:
+        return False
+
+
+def get_id(node, is_leaf=True):
+    """
+    Get identifier if defined.
+    """
+    actual_node = id_location(node)
+    if is_leaf and (dereference_node(node).name != actual_node.name):
```

Comment:
Should `is_id_defined_in_leaf()` be used here as well?

Reply:
should be, just relict of old code